aopy/analysis/base.py

```diff
@@ -1621,3 +1621,42 @@ def calc_corr2_map(data1, data2, knlsz=15, align_maps=False):
     NCC[nan_idx1] = np.nan
     
     return NCC, shifts
+
+def get_neighboring_channel_index(elec_pos, distance=0.75):
+    '''
+    Get neiboring channel index based on distance
+    
+    Args:
+        elec_pos (nch, 2): x, y position of each channel
+        distance (float): define neighboring channels based on this distance. 0.75 is for 4 neighboring channels and 1.3 is for 9 channels.
+        
+    Returns:
+        neighboring_idx (nch, nch): boolean array to show neighboring channels for each channel
+    '''
+    dist = utils.calc_euclid_dist_mat(elec_pos)
+    neighboring_idx = dist <= distance
+        
+    return neighboring_idx
+
+def get_mean_neighboring_channels(data, elec_pos, distance=0.75):
```

Comment:
i don't want to derail this function -- it's fine and you can push it -- but just a note for the future: we should really make a spatial smoothing function that takes an arbitrary 2d kernel and smooths the data that way, rather than only doing average and only limited to a radius